src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/CallStats.cs

```diff
@@ -13,26 +13,33 @@ namespace SharedPoolsOfWCFObjects
     /// <summary>
     /// This class provides a few Call*AndRecordStats() utility methods which will store and aggregate the timing of the delegates they invoke 
     /// 
-    /// Note: all Call*AndRecordStats() methods catch all exceptions coming from the delegates
-    /// However we don't want to miss the case when an unexpected exception happens for the calls that should never have exceptions
+    /// Note that there are 2 separate mechanisms to control how the exceptions are handled by these utility methods.
+    /// The first one is the boolean "hideExceptions" parameter to all Call*AndRecordStats methods where 'true' means completely ignore exceptions.
+    /// This is used by the scenarios where the exceptions are expected and should be ignored no matter what kind of test is running.
+    /// 
+    /// The second one is the Func<Exception, bool> exceptionHandler parameter which is typically controlled by the caller via generic TestParams.
+    /// This allows different level of tolerance for different test runs (e.g. no unexpected exceptions for Perf and some tolerance for Stress).
+    /// The boolean return value signifies if the exception is considered to be "handled" - e.g. no need to propagate the exception to the caller.
     /// </summary>
     public class CallStats
     {
+        public Func<Exception, bool> _exceptionHandler;
         public CallTimingStatsCollector SunnyDay { get; set; }
         public CallTimingStatsCollector RainyDay { get; set; }
 
-        public CallStats(int samples, int errorSamples)
-            : this(samples, errorSamples, CallTimingStatsCollector.CalculateSomePercentiles, CallTimingStatsCollector.CalculateSomePercentiles)
+        public CallStats(int samples, int errorSamples, Func<Exception, bool> exceptionHandler)
+            : this(samples, errorSamples, exceptionHandler, CallTimingStatsCollector.CalculateSomePercentiles, CallTimingStatsCollector.CalculateSomePercentiles)
         {
         }
 
-        public CallStats(int samples, int errorSamples, Func<long[], TimingPercentileStats> statsCalculator, Func<long[], TimingPercentileStats> errorStatsCalculator)
+        public CallStats(int samples, int errorSamples, Func<Exception, bool> exceptionHandler, Func<long[], TimingPercentileStats> statsCalculator, Func<long[], TimingPercentileStats> errorStatsCalculator)
         {
             SunnyDay = new CallTimingStatsCollector(samples, statsCalculator);
             RainyDay = new CallTimingStatsCollector(errorSamples, errorStatsCalculator);
+            _exceptionHandler = exceptionHandler;
```

Comment:
>exceptionHandler [](start = 32, length = 16)

Nit: not required to merge this PR -- if a null exceptionHandler is passed in, we will get NullReferenceException trying to use it in the tests below.  I know your tests always pass it in, but it would be a good idea to Contract.Assert() it is not null (if built Debug) or throw an ArgumentException (if Retail).  Alternatively, if a null is passed in, silently set _exceptionHandler to a func that returns true.  Recommended only for future consideration -- don't make a new PR just for this.